### ISSUE TITLE
[BUG] Lua scripts not working anymore after update to 25.0.7 (macOS)

### ISSUE THREAD

**Issue by milanvanschaik** [obsproject/obs-studio#2784]
## Platform

Operating system and version: macOS 10.15.3
OBS Studio version: 25.0.7

## Expected Behavior

Lua scripts should work and did before correctly.

## Current Behavior

After updating to OBS 25.0.7 I get the following message when importing a Lua script:
```
Failed to create new lua state
```

## Steps to Reproduce

1. Make sure to be on OBS 25.0.7 on macOS (10.15.3?)
2. Import one of the example scripts, for example the countdown
3. You get the `Failed to create new lua state` message and won't have a working script

## Additional information

Great work on the new version. I got no more dropping frames at all on a 1080p canvas and everything is butter smooth as far as I noticed, except for the Lua scripts where my productions depend on.

**Comment by PatTheMav** [comment]
I can't reproduce this with 25.0.7 on Catalina right now, the countdown works as expected.

Would you be able to attach a log of a session in which the error occurs?

**Comment by milanvanschaik** [comment]
@PatTheMav Here you go!
```
14:45:38.783: CPU Name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
14:45:38.783: CPU Speed: 2600MHz
14:45:38.783: Physical Cores: 4, Logical Cores: 8
14:45:38.783: Physical Memory: 16384MB Total
14:45:38.783: OS Name: Mac OS X (NSMACHOperatingSystem)
14:45:38.783: OS Version: Version 10.15.3 (Build 19D76)
14:45:38.783: Kernel Version: 19.3.0
14:45:38.786: hotkeys-cocoa: Getting keyboard keys failed
14:45:38.786: hotkeys-cocoa: Getting keyboard keys failed
14:45:38.786: hotkeys-cocoa: Using layout 'com.apple.keylayout.Dutch'
14:45:38.787: Portable mode: false
14:45:38.952: OBS 25.0.7 (mac)
14:45:38.952: ---------------------------------
14:45:38.953: ---------------------------------
14:45:38.953: audio settings reset:
14:45:38.953: 	samples per sec: 44100
14:45:38.953: 	speakers:        2
14:45:38.957: ---------------------------------
14:45:38.957: Initializing OpenGL...
14:45:42.782: Loading up OpenGL on adapter ATI Technologies Inc. AMD Radeon Pro 450 OpenGL Engine
14:45:42.782: OpenGL loaded successfully, version 4.1 ATI-3.5.5, shading language 4.10
14:45:42.972: ---------------------------------
14:45:42.972: video settings reset:
14:45:42.972: 	base resolution:   1920x1080
14:45:42.972: 	output resolution: 1920x1080
14:45:42.972: 	downscale filter:  Bicubic
14:45:42.972: 	fps:               30/1
14:45:42.972: 	format:            NV12
14:45:42.972: 	YUV mode:          601/Partial
14:45:42.973: NV12 texture support not available
14:45:42.976: Audio monitoring device:
14:45:42.976: 	name: Default
14:45:42.976: 	id: default
14:45:42.977: ---------------------------------
14:45:42.986: Failed to load 'en-US' text for module: 'decklink-ouput-ui.so'
14:45:43.077: os_dlopen(libpython3.7m.dylib->libpython3.7m.dylib): dlopen(libpython3.7m.dylib, 257): image not found
14:45:43.077: 
14:45:43.077: [Python] Could not load library: libpython3.7m.dylib
14:45:43.412: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
14:45:43.412: No blackmagic support
14:45:43.611: [VideoToolbox encoder]: Adding VideoToolbox H264 encoders
14:45:43.784: [obs-browser]: Version 2.8.6
14:45:43.858: VLC found, VLC video source enabled
14:45:43.863: [obs-ndi] hello ! (version 4.7.1)
14:45:43.863: [obs-ndi] Trying ''
14:45:43.863: [obs-ndi] Trying '/usr/lib'
14:45:43.863: [obs-ndi] Trying '/usr/local/lib'
14:45:43.863: [obs-ndi] Found NDI library at '/usr/local/lib/libndi.4.dylib'
14:45:43.868: [obs-ndi] NDI runtime loaded successfully
14:45:43.869: [obs-ndi] NDI library initialized successfully (NDI SDK APPLE 14:35:05 Dec 19 2019 4.1.3)
14:45:43.875: ---------------------------------
14:45:43.875:   Loaded Modules:
14:45:43.875:     obs-ndi.so
14:45:43.875:     vlc-video.so
14:45:43.875:     text-freetype2.so
14:45:43.875:     rtmp-services.so
14:45:43.875:     obs-x264.so
14:45:43.875:     obs-vst.so
14:45:43.875:     obs-transitions.so
14:45:43.875:     obs-outputs.so
14:45:43.875:     obs-libfdk.so
14:45:43.875:     obs-filters.so
14:45:43.875:     obs-ffmpeg.so
14:45:43.875:     obs-browser.so
14:45:43.875:     mac-vth264.so
14:45:43.875:     mac-syphon.so
14:45:43.875:     mac-decklink.so
14:45:43.875:     mac-capture.so
14:45:43.875:     mac-avcapture.so
14:45:43.875:     linux-jack.so
14:45:43.875:     image-source.so
14:45:43.875:     frontend-tools.so
14:45:43.875:     decklink-ouput-ui.so
14:45:43.875:     coreaudio-encoder.so
14:45:43.875: ---------------------------------
14:45:43.875: os_dlopen(../obs-plugins/obs-browser->../obs-plugins/obs-browser.so): dlopen(../obs-plugins/obs-browser.so, 257): image not found
14:45:43.875: 
14:45:43.876: ==== Startup complete ===============================================
14:45:43.913: All scene data cleared
14:45:43.913: ------------------------------------------------
14:45:43.962: [Lua: countdown.lua] Failed to create new lua state
```

**Comment by PatTheMav** [comment]
It's possibly related to this luajit issue, which would require an update of the library (from luajit 2.0.5 to 2.1.0) OBS is built with: https://github.com/LuaJIT/LuaJIT/issues/521.

Version 2.1.0 is not stable however, so a judgement call is necessary - ping @DDRBoxman

**Comment by PatTheMav** [comment]
@milanvanschaik I've compiled a new version of the obs-scripting plugin with an updated luajit version - as I can't reproduce the issue, would you be willing to update these two files in a local copy of OBS-Studio on your Mac and check if it fixes your issue?

**Comment by jvincilione** [comment]
@PatTheMav I'm having the same issue and would be happy to test this. Can you point me to the instructions on standing this up locally?

**Comment by PatTheMav** [comment]
@jvincilione You could of course checkout the patched `obs-deps` from my draft PR above, then compile OBS-Studio yourself pointing it to those dependencies.

`obs-deps` has a single build script you can run (found at `CI/build-package-deps-osx.sh`), then after that you can setup your OBS build by adding `-DDepsPath=/tmp/obsdeps` to the `cmake` step outlined here: https://github.com/obsproject/obs-studio/wiki/install-instructions#macos-build-directions

**Comment by jvincilione** [comment]
@PatTheMav Not sure what the issue is, but it just errors when I try to build. I'll try on my other machine later if no one else can get to it.

**Comment by DDRBoxman** [comment]
Should be fixed in 25.0.8, let me know if otherwise.

Thanks for the fix @PatTheMav :)
